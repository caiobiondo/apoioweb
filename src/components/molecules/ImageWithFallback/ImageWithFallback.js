import React from 'react';
import Img from 'react-image';
import { CircularProgress, Icon } from 'natura-ui';
import { Wrapper } from './ImageWithFallback.styles';

const ImageWithFallback = props => {
  const fallbackImage = props.unloader ? props.unloader : <Icon file={props.fallbackIcon} />;
  const loader = <CircularProgress thickness={2} />;

  return (
    <Wrapper width={props.width} height={props.height || props.width} className={props.className}>
      <Img src={props.imageUrl} loader={loader} unloader={fallbackImage} />
    </Wrapper>
  );
};

ImageWithFallback.defaultProps = {
  fallbackIcon: 'ico_pictureless',
};

export default ImageWithFallback;
