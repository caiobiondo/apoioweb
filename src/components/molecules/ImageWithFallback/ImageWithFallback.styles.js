import styled from 'styled-components';
import { gray200 } from 'styles/colors';

export const Wrapper = styled.div`
  svg,
  img {
    width: ${props => props.width};
    height: ${props => props.height};
  }

  svg {
    fill: ${gray200};
  }
`;
