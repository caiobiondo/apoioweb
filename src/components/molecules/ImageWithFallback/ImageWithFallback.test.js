import React from 'react';
import { shallow } from 'enzyme';
import ImageWithFallback from './ImageWithFallback';

describe('ImageWithFallback', () => {
  it('should render correctly', () => {
    const result = shallow(
      <ImageWithFallback
        className="test-class"
        fallbackIcon="ico_fallback"
        height="10px"
        imageUrl="http://www.test.com/image.jpg"
        width="10px"
      />,
    );
    expect(result).toMatchSnapshot();
  });

  it('should render correctly (with only width)', () => {
    const result = shallow(
      <ImageWithFallback
        className="test-class"
        fallbackIcon="ico_fallback"
        imageUrl="http://www.test.com/image.jpg"
        width="10px"
      />,
    );
    expect(result).toMatchSnapshot();
  });
});
