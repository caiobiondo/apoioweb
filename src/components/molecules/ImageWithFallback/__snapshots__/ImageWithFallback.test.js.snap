// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWithFallback should render correctly (with only width) 1`] = `
<ImageWithFallbackstyles__Wrapper
  className="test-class"
  height="10px"
  width="10px"
>
  <Img
    decode={true}
    loader={
      <CircularProgress
        max={100}
        min={0}
        mode="indeterminate"
        size={40}
        thickness={2}
        value={0}
      />
    }
    src="http://www.test.com/image.jpg"
    unloader={
      <Icon
        file="ico_fallback"
      />
    }
  />
</ImageWithFallbackstyles__Wrapper>
`;

exports[`ImageWithFallback should render correctly 1`] = `
<ImageWithFallbackstyles__Wrapper
  className="test-class"
  height="10px"
  width="10px"
>
  <Img
    decode={true}
    loader={
      <CircularProgress
        max={100}
        min={0}
        mode="indeterminate"
        size={40}
        thickness={2}
        value={0}
      />
    }
    src="http://www.test.com/image.jpg"
    unloader={
      <Icon
        file="ico_fallback"
      />
    }
  />
</ImageWithFallbackstyles__Wrapper>
`;
